src/formatters/junitFormatter.ts

```diff
@@ -76,6 +76,24 @@ export class Formatter extends AbstractFormatter {
             }
         }
 
+        if (fileNames !== undefined && fileNames.length !== 0) {
+            // Filter out files which have had a failure associated with them.
+            const filteredFileNames = fileNames.filter(fileName => {
+                for (const failure of failures) {
+                    if (fileName === failure.getFileName()) {
```

Comment:
This loops over all of `failures` for each `fileName`, so its runtime complexity is slightly better than `O(fileNames.length * failures.length)`, but still at worst a larger-than-necessary loop.

It'd be more efficient to create some kind of mapping between file names and their failures. How about something like a `Set<string>` of file names that _do_ have failures? That'll change your runtime complexity to roughly `O(fileNames.length + failures.length)`.

Normally the performance & runtime complexities for a method that runs once to format something wouldn't be too important, but in this case it might be called with thousands of file names and hundreds of failures.

Reply:
No worries! Added something that's a bit more optimised using a set as you suggest. Will add the new unit test ASAP.